docs/interfaces.md

```diff
@@ -0,0 +1,591 @@
+                SGX_EPID = 0;
+                SGX_DCAP = 1;
+                WHATEVER = 2;
+        }
+        Type type     = 1;
+        bytes payload = 2;
+}
+
+```
+
+ERCC keeps in instance of an attestation.Verifier to check an attestation evidence message. ERCC just passes the serialized attestation evidence message to the verifier. 
+Depending on the attestation protocol (e.g., EPID- or DCAP-based attestation), the verifier implements the corresponding logic. Details of the evidence verification are defined in [#412](https://github.com/hyperledger-labs/fabric-private-chaincode/issues/412).
+
+```go
+type EnclaveRegistryCC struct {
+	ra  attestation.Verifier
+}
+```
+
+
+# TLCC
+
+Some notes on TLCC. The TLCC instance is currently implemented as a system chaincode, that is, there exists only a single instance. As a peer can participate in many channels, there is a separate TLCC_enclave per each channel. The TLCC chaincode is responsible to multiplex requests for a given channel to the corresponding TLCC_enclave. Note for MVP, we only support a single channel but the interface specified here should already take multi-channel support into account.
+
+## Interface:
+The TLCC interface is implemented through a "normal" chaincode interface.
+
+```go
+// chaincode interface
+func Invoke(stub shim.ChaincodeStubInterface) pb.Response {}
+
+// functions below are dispatched by Invoke implementation
+
+// used by the admin to join the channel
+// Note: this methods creates a new tlcc_enclave and starts a block listener
+// for the given channel_id that passes the genesis block and all
+// subsequent blocks to the tlcc_enclave.
+func joinChannel(channel_id string) error {}
+
+// returns the SHA256 hash of the channel genesis block
+// the tlcc enclave is initialized with
+func getChannelHash(channel_id string) (string, error) {}
+```
+
+```c++
+// Provided/implemented by a common logging module
+// interface exposed to TLCC_enclave via cgo
+void ocall_print_string([in, string] const char *str);
+```
+
+```go
+// interface to establish/close a secure channel between an
+// ECC_Enclave and a TLCC_enclave.
+func session_setup(setup SessionSetup) (SessionSetupResponse, error) {}
+func session_close(close SessionClose) (SessionCloseResponse, error) {}
+
+// interface to invoke a TLCC request, such as, getMetadata and validateMSPId
+func session_request(request SessionRequest) (SessionRequestResponse, error) {}
+```
+
+## State:
+
+```protobuf
+syntax = "proto3"
+
+// Setup and Close messages are used to establish a secure channel between
+// ECC_Enclave and TLCC_Enclave. Once a secure channel is established,
+// Session Requests to TLCC can be issued to invoke TLCC functions such
+// as getMetadata or verifyMSPId.
+
+// TODO How to integrate the the message authenticator?!
+// should we abstract it as signature here even though we may
+// use CMAC once a session is established?
+
+message SessionSetup {
+        string channel_id = 1;
```

Comment:
This should also be channel_hash. I think internally in our (enclave) code, we should use only channel_hash and never channel_id, i.e., channel_id only as needed to find channel_hash.  Insofar, if possible i would prefer to have channel_hash in CC_Params (and no reason then to have channel_hash in credentials)?